core/server/data/validation/index.js

```diff
@@ -42,6 +43,27 @@ validator.extend('isTimezone', function isTimezone(str) {
     return moment.tz.zone(str) ? true : false;
 });
 
+validator.extend('isEmptyOrWebsite', function isEmptyOrWebsite(str) {
```

Comment:
Rather than strict validation that can break perfectly valid URLs I think we should rather store URLs in a safer format. If we urlencode the value before saving then that avoids the problem of output not being sanitised before being rendered. Javascript even has a built in method for it 😉 

<img width="541" alt="screen shot 2017-01-06 at 17 30 48" src="https://cloud.githubusercontent.com/assets/415/21726763/f8810bb0-d435-11e6-9725-9bcf281266b2.png">